### PR TITLE
Refactor for end-to-end tests

### PR DESCRIPTION
**Pull request by msilvagarcia**
* Improve additional details handling
* Make tests easier
* Move variable instantiation

### CODE REVIEW

**Comment by rikterbeek** [comment]
@msilvagarcia can you fix the tests ?

File: controllers/front/Validate3d.php [hunk 1]
```diff
@@ -87,11 +87,24 @@ public function postProcess()
...
                     $payment = $new_order->getOrderPaymentCollection();
                     if (isset($payment[0])) {
-                        //todo add !empty
-                        $payment[0]->card_number = pSQL($response['additionalData']['cardBin'] . " *** " . $response['additionalData']['cardSummary']);
-                        $payment[0]->card_brand = pSQL($response['additionalData']['paymentMethod']);
-                        $payment[0]->card_expiration = pSQL($response['additionalData']['expiryDate']);
-                        $payment[0]->card_holder = pSQL($response['additionalData']['cardHolderName']);
+                        $cardSummary = !empty($response['additionalData']['cardSummary'])
+                            ? pSQL($response['additionalData']['cardSummary'])
+                            : '****';
```
Comment (cyattilakiss) [inline controllers/front/Validate3d.php]: I like this approach we should do the same in M2 and other plugins as well

**Comment by msilvagarcia** [comment]
@rikterbeek Travis seem to be reading an old file for the push builds. The Pull Request build is passing and using the correct .travis.yml

**Comment by msilvagarcia** [comment]
Ah, figured it out! :D

**Pull request merged**